sample-apps/react-native/dogfood/src/components/LivestreamChat.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from 'react';
+import { Platform, SafeAreaView, StyleSheet } from 'react-native';
+import {
+  Channel,
+  MessageInput,
+  MessageList,
+  useChatContext,
+} from 'stream-chat-react-native';
+import { AuthenticationProgress } from '../components/AuthenticatingProgress';
+import { Channel as ChannelType } from 'stream-chat';
+import { StreamChatGenerics } from '../../types';
+import { useBottomSheetInternal } from '@gorhom/bottom-sheet';
+import { appTheme } from '../theme';
+
+export type LivestreamChatProps = {
+  callId: string;
+  callType: string;
+};
+
+export const LivestreamChat = ({ callId, callType }: LivestreamChatProps) => {
+  const [channel, setChannel] = useState<
+    ChannelType<StreamChatGenerics> | undefined
+  >(undefined);
+  const { client } = useChatContext();
+  const { shouldHandleKeyboardEvents } = useBottomSheetInternal();
+
+  // Done as per https://ui.gorhom.dev/components/bottom-sheet/keyboard-handling/ to solve the issue around keyboard hiding the text input in the chat inside bottom sheet.
+  useEffect(() => {
+    return () => {
+      // Reset the flag on unmount
+      shouldHandleKeyboardEvents.value = false;
+    };
+  }, [shouldHandleKeyboardEvents]);
+
+  useEffect(() => {
+    const createChannel = async () => {
+      const newChannel = await client.channel(callType, callId);
+      setChannel(newChannel);
+    };
+    createChannel();
+  }, [client, callId, callType]);
+
+  if (!channel) {
+    return (
+      <SafeAreaView
+        style={[
+          styles.container,
+          { backgroundColor: appTheme.colors.static_grey },
+        ]}
+      >
+        <AuthenticationProgress />
+      </SafeAreaView>
+    );
+  }
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <Channel
+        additionalTextInputProps={
+          Platform.OS === 'ios'
```

Comment:
why only for iOS though? in bottomsheet lib its done for both platforms

Reply:
In Android the behaviour is weird with this. On Android everything works well by default for our chat SDK text input.

Reply:
![Screenshot_1712555407](https://github.com/GetStream/stream-video-js/assets/39884168/368a8c48-0857-4fd6-81b7-ba231b4e2a5f)

Reply:
then better to add the comment that in Android its unnecessary to handle

Reply:
Pushed 👍🏻